### PR TITLE
refactor(client/dead): setting downed states

### PR DESCRIPTION
**Pull request by Manason**
**Describe Pull request**
- extracted functions from thread which handles last stand and dead states
- replaced loadAnimDict local function with calls to ox_lib
- moved thread and dependent local functions to new file

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [no] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

File: client/dead.lua [hunk 1]
```diff
@@ -125,104 +96,4 @@ AddEventHandler('gameEventTriggered', function(event, data)
...
-                            StopAnimTask(ped, lastStandDict, lastStandAnim, 3)
-                        end
-                    end
-                end
-            end
-        end
-        Wait(sleep)
-    end
-end)
+emsNotified = false
```
Comment (BerkieBb) [inline client/dead.lua]: Why is this at the bottom?

File: client/setdownedstate.lua [hunk 2]
```diff
@@ -0,0 +1,150 @@
+local deadAnimDict = "dead"
+local deadAnim = "dead_a"
+
+local function drawTxt(x, y, width, height, scale, text, r, g, b, a, _)
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: The last underscore is useless, you can remove it

File: client/setdownedstate.lua [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    SetTextEntry("STRING")
+    AddTextComponentString(text)
+    DrawText(x - width / 2, y - height / 2 + 0.005)
+end
+
+local function isEmsOnDuty()
+    QBCore.Functions.TriggerCallback('hospital:GetDoctors', function(medics)
+        return medics > 0
+    end)
+end
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: This won't actually return in the isEmsOnDuty function, but rather just end the execution of the callback, so this is the fix
```suggestion
local function isEmsOnDuty()
    local p = promise.new()
    QBCore.Functions.TriggerCallback('hospital:GetDoctors', function(medics)
        p:resolve(medics > 0)
    end)

    return Citizen.Await(p)
end
```

File: client/setdownedstate.lua [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    if deathTime > 0 and isEmsOnDuty() then
+        drawTxt(0.93, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_txt', { deathtime = math.ceil(deathTime) }), 255, 255, 255, 255)
+    else
+        drawTxt(0.865, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_revive', { holdtime = hold, cost = Config.BillCost }), 255, 255, 255, 255)
+    end
+end
+
+local function playDeadAnimation(ped)
+    if IsPedInAnyVehicle(ped, false) then
+
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: weird empty space

File: client/setdownedstate.lua [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+local function displayRespawnText()
+    if deathTime > 0 and isEmsOnDuty() then
+        drawTxt(0.93, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_txt', { deathtime = math.ceil(deathTime) }), 255, 255, 255, 255)
+    else
+        drawTxt(0.865, 1.44, 1.0, 1.0, 0.6, Lang:t('info.respawn_revive', { holdtime = hold, cost = Config.BillCost }), 255, 255, 255, 255)
+    end
+end
+
+local function playDeadAnimation(ped)
+    if IsPedInAnyVehicle(ped, false) then
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: replace with cache.vehicle

File: client/setdownedstate.lua [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    if not isInHospitalBed then
+        displayRespawnText()
+    end
+
+    playDeadAnimation(ped)
+    SetCurrentPedWeapon(ped, `WEAPON_UNARMED`, true)
+end
+
+local function playEscortedLastStandAnimation(ped)
+    if IsPedInAnyVehicle(ped, false) then
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: replace with cache.vehicle

File: client/setdownedstate.lua [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    else
+        lib.requestAnimDict(lastStandDict)
+        if IsEntityPlayingAnim(ped, lastStandDict, lastStandAnim, 3) then
+            StopAnimTask(ped, lastStandDict, lastStandAnim, 3)
+        end
+    end
+end
+
+local function playUnescortedLastStandAnimation(ped)
+    if IsPedInAnyVehicle(ped, false) then
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: replace with cache.vehicle

File: client/setdownedstate.lua [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+    local ped = PlayerPedId()
+    while true do
+        if isDead or InLaststand then
+            disableControls()
+            if isDead then
+                handleDead(ped)
+            elseif InLaststand then
+                handleLastStand(ped)
+            end
+            Wait(5)
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: 0 not 5

File: client/setdownedstate.lua [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    EnableControlAction(0, 288, true)
+    EnableControlAction(0, 213, true)
+    EnableControlAction(0, 249, true)
+    EnableControlAction(0, 46, true)
+    EnableControlAction(0, 47, true)
+end
+
+--- Set dead and last stand states.
+CreateThread(function()
+    local ped = PlayerPedId()
```
Comment (BerkieBb) [inline client/setdownedstate.lua]: use cache.ped instead of the ped variable overall because this won't work anymore if the ped changes

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: client/dead.lua (hunk 1 above)
Comment (Manason) [reply client/dead.lua]: moved to the top.

**Review by BerkieBb — APPROVED** [review]

**Review by YvanCywan — APPROVED** [review]

**Pull request merged**